From cf4fa033afbd0506adaa9dcf23da5f7651ce73a3 Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Fri, 25 Apr 2014 10:58:17 +0800
Subject: [PATCH 1/2] ENGR00306778 - NFS: Upgrading process is shown up when
 NFS boot up. 100%

When NFS boot, the node's(except those created on memory such as /proc, /dev)
xattr is all set to u:object_r:rootfs:s0. Which is not excepted by Android.
Then system server restart in another process, since /data/system/packages.xml
is creatted before, together with other conditions, it's thought as upgrade.
If NFS boot, thought as selinux is disabled can fix it.

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 Android.mk    | 1 +
 src/enabled.c | 8 ++++++++
 2 files changed, 9 insertions(+)

diff --git a/Android.mk b/Android.mk
index 6423177..4e9e2b1 100644
--- a/Android.mk
+++ b/Android.mk
@@ -70,4 +70,5 @@ LOCAL_MODULE_TAGS := eng
 LOCAL_COPY_HEADERS_TO := $(common_COPY_HEADERS_TO)
 LOCAL_COPY_HEADERS := $(common_COPY_HEADERS)
 LOCAL_PRELINK_MODULE := false
+LOCAL_SHARED_LIBRARIES := libcutils
 include $(BUILD_SHARED_LIBRARY)
diff --git a/src/enabled.c b/src/enabled.c
old mode 100644
new mode 100755
index 569a7a0..6f9b569
--- a/src/enabled.c
+++ b/src/enabled.c
@@ -7,6 +7,7 @@
 #include <limits.h>
 #include <stdio.h>
 #include "policy.h"
+#include "cutils/properties.h"
 
 int is_selinux_enabled(void)
 {
@@ -15,6 +16,13 @@ int is_selinux_enabled(void)
 	char *bufp;
 	int enabled = 0;
 	security_context_t con;
+	char nfsModeValue[PROP_VALUE_MAX];
+	int ret;
+
+	ret = property_get("ro.nfs.mode", nfsModeValue, NULL);
+	if((ret ==  3) && (strncmp(nfsModeValue, "yes", 3) == 0)) {
+		return 0;
+	}
 
 	/* init_selinuxmnt() gets called before this function. We
  	 * will assume that if a selinux file system is mounted, then
-- 
1.8.0

